internal/storage/v2/grpc/factory.go

```diff
@@ -52,17 +52,24 @@ func NewFactory(
 		config: cfg,
 	}
 
+	var writerConfig *configgrpc.ClientConfig
```

Comment:
Any reason not to copy by value? Sharing could have side effects.
```suggestion
	var writerConfig configgrpc.ClientConfig
```